File: app/src/main/java/com/example/i1/clicker/Saver.java
package com.example.i1.clicker;

import android.content.Context;
import android.content.SharedPreferences;

import static android.content.Context.MODE_PRIVATE;

/**
 * Created by morozione on 07/19/17.
 */

public class Saver {

    public static final String KEY_RECORD = "record";
    public static final String KEY_COINS = "coins";
    public static final String KEY_COUNT_MINUTES = "count_minutes";

    private SharedPreferences dataRecord;
    private static Saver saver;

    private Saver(Context context) {
        dataRecord = context.getSharedPreferences(KEY_RECORD, MODE_PRIVATE);
    }

    public static Saver getInstance(Context context) {
        if (saver == null)
            saver = new Saver(context);
        return saver;
    }

    public void saveInt(int record, String key) {
        SharedPreferences.Editor edit = dataRecord.edit();
        edit.putInt(key, record);
        edit.commit();
    }

    public int loadInt(String key) {
        int record = dataRecord.getInt(key, 0);
        return record;
    }
}

File: app/src/main/java/com/morozione/clicker/Saver.kt
package com.morozione.clicker

import android.content.Context
import android.content.SharedPreferences
import androidx.core.content.edit
import dagger.hilt.android.qualifiers.ApplicationContext
import javax.inject.Inject
import javax.inject.Singleton

@Singleton
class Saver @Inject constructor(
    @ApplicationContext context: Context
) {
    private val dataRecord: SharedPreferences = context.getSharedPreferences(KEY_RECORD, Context.MODE_PRIVATE)

    fun saveInt(value: Int, key: String) {
        dataRecord.edit {
            putInt(key, value)
        }
    }

    fun loadInt(key: String, defaultValue: Int = 0): Int =
        dataRecord.getInt(key, defaultValue)

    companion object {
        const val KEY_RECORD = "record"
        const val KEY_COINS = "coins"
        const val KEY_COUNT_MINUTES = "count_minutes"
    }
} 